Gems/EMotionFX/Code/Source/Editor/Plugins/SkeletonOutliner/JointPropertyWidget.cpp

```diff
@@ -111,4 +154,185 @@ namespace EMotionFX
+        m_filterString = m_filterEntityBox->text().toLatin1().data();
+
+        m_clothJointWidget->SetFilterString(m_filterString);
+        m_hitDetectionJointWidget->SetFilterString(m_filterString);
+        m_ragdollJointWidget->SetFilterString(m_filterString);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    //
+    //    AddCollidersButton
+    //
+    AddCollidersButton::AddCollidersButton(QWidget *parent)
+        :QPushButton(parent)
+    {
+        setText("Add Collider");
+        connect(this, &QPushButton::clicked, this, &AddCollidersButton::OnCreateContextMenu);
+    }
+
+    // TODO subclass ItemModel
+    enum ItemRoles
+    {
+        TypeId = Qt::UserRole + 1,
+        ConfigType = Qt::UserRole + 2
+    };
+    void AddCollidersButton::OnCreateContextMenu()
+    {
+        delete model;
+        model = new QStandardItemModel;
+        QFrame* newFrame = new QFrame{this};
+        newFrame->setWindowFlags(Qt::FramelessWindowHint | Qt::Popup);
+        newFrame->setFixedWidth(this->width());
+        newFrame->move({ this->mapToGlobal({ 0, 0 }) });
+        auto* treeView = new QTreeView(newFrame);
+        treeView->setModel(model);
+        // hide header for dropdown-style, single-column, tree
+        treeView->header()->hide();
+        connect(treeView, &QTreeView::clicked, this, &AddCollidersButton::OnAddColliderActionTriggered);
+        treeView->setEditTriggers(QAbstractItemView::NoEditTriggers);
+        newFrame->setLayout(new QVBoxLayout);
+        newFrame->layout()->addWidget(treeView);
+
+        AZStd::string actionName;
+        auto sections = QList<QPair<PhysicsSetup::ColliderConfigType, QString>>{{PhysicsSetup::ColliderConfigType::Cloth, "Cloth"},
+                         {PhysicsSetup::ColliderConfigType::HitDetection, "Hit Detection"}};
+        for (const auto& s : sections)
+        {
+            auto& configType = s.first;
+            auto* sectionItem = new QStandardItem{ s.second.toStdString().c_str() };
+
+            for (auto& typeId : m_supportedColliderTypes)
+            {
+                actionName = AZStd::string::format("Add %s %s", s.second.toStdString().c_str(), GetNameForColliderType(typeId).c_str());
+                auto* item = new QStandardItem{ actionName.c_str() };
+                item->setData(typeId.ToString<AZStd::string>().c_str(), ItemRoles::TypeId);
+                item->setData(static_cast<int>(configType), ItemRoles::ConfigType);
+                sectionItem->appendRow(item);
+            }
+            model->appendRow(sectionItem);
+        }
+
+        auto ragdollItem = new QStandardItem{ "Ragdoll" };
+        model->appendRow(ragdollItem);
+        ragdollItem->setData(PhysicsSetup::ColliderConfigType::Ragdoll, ItemRoles::ConfigType);
+
+        // todo copy from other collider type
+        // todo paste (if there is a copied collider)
```

Comment:
update: Remove todo comments (I know this is likely here as it's a WIP review but they have a habit of getting forgotten about 😅)